ext/opcache/Optimizer/pass1_5.c

```diff
@@ -309,6 +307,11 @@ void zend_optimizer_pass1(zend_op_array *op_array, zend_optimizer_ctx *ctx)
 						}
 					}
 				} else if (op_array->scope &&
+					ZEND_OP1_TYPE(opline) == IS_UNUSED &&
+					(opline->op1.num & ~ZEND_FETCH_CLASS_NO_AUTOLOAD) == ZEND_FETCH_CLASS_SELF) {
```

Comment:
I mean that I suspect (but haven't verified) that this optimization (both in this branch and previously) currently doesn't trigger, because we check for `(num & ~ZEND_FETCH_CLASS_NO_AUTOLOAD) == ZEND_FETCH_CLASS_SELF`, while `num` will also contain a `ZEND_FETCH_CLASS_EXCEPTION` flag, so the check will never be true.